File: 백준/Silver/5567. 결혼식/결혼식.cc
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

int n,m;

vector<int> v[501];
bool visited[501];
int answer;

void find_friends(int x){
  visited[x] = true;
  queue<pair<int,int>> q; // (번호, 깊이)
  q.push({x,0});
  
  while(!q.empty()){
    auto tmp = q.front();
    int num = tmp.first;
    int d = tmp.second;
    q.pop();
    if(d >= 2) continue;
    
    for(int i=0 ; i<v[num].size(); i++){
      int num2 = v[num][i];
      if(visited[num2]) continue;
      
      q.push({num2,d+1});
      visited[num2] = true;
      answer++;
    }
  }

}

int main() 
{
    cin >> n >> m;
    int a,b;
    for(int i=0 ; i<m; i++){
      cin >> a >> b;
      v[a].push_back(b);
      v[b].push_back(a);
    }
    
    find_friends(1);
    
    cout << answer;
    
    return 0;
}

File: 백준/Silver/5567. 결혼식/결혼식.java
import java.util.*;
import java.io.*;

public class Main {
    static int n,m;
    static List<Integer>[] v;
    static boolean[] visited;
    static int answer;
    
    static void findFriends(int x){
      visited[x] = true;
      Queue<int[]> q = new ArrayDeque<>(); // (번호, 깊이) 저장
      q.offer(new int[]{x,0});
      
      while(!q.isEmpty()){
        int[] top = q.poll();
        int num = top[0];
        int d = top[1];
        
        if(d>=2) continue;
        
        for(int i=0 ; i<v[num].size(); i++){
          int nx = v[num].get(i); // ArrayList 원소에 접근
          if(visited[nx] == true) continue;
          
          visited[nx] = true;
          q.offer(new int[]{nx,d+1});
          answer++;
        }
      }
    }
  
    public static void main(String[] args) throws IOException {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      StringTokenizer st = new StringTokenizer(br.readLine());
      
      n = Integer.parseInt(st.nextToken());
      st = new StringTokenizer(br.readLine());
      m = Integer.parseInt(st.nextToken());
      
      v = new ArrayList[n+1];
      visited = new boolean[n+1];
      for(int i=0 ; i<=n; i++){
        v[i] = new ArrayList<>();
      }
      
      int a,b;
      for(int i=0 ; i<m ; i++){
        st = new StringTokenizer(br.readLine());
        a = Integer.parseInt(st.nextToken());
        b = Integer.parseInt(st.nextToken());
        
        v[a].add(b);
        v[b].add(a);
      }
      
      findFriends(1);
      
      System.out.println(answer);
    }
}